src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs

```diff
@@ -494,14 +495,19 @@ private void ReadPdhMultiString(ref IntPtr strNative, Int32 strSize, ref StringC
                     break;
                 }
 
-                allSubstringsWithNulls += (char)next4;
+                char nextChar = (char)next4;
+                if (nextChar == '\0')
+                {
+                    strColl.Add(sb.ToString());
```

Comment:
Adding to strColl should occur outside the while loop. The updated logic causes the contents of sb to be dropped on the floor by the break statement when next4 == 0.

Essentially, replace the original `allSubstringsWithNulls += (char)next4` statement with `sb.Append(nextChar) `and move the `strColl.Add` statement to the original location after the while loop.

Reply:
You're right, but adding to strColl only outside the while loop has some memory implications (has to alloc at least one extra string before Split). Can we just add another `strColl.Add(sb.ToString());` outside the while to catch whatever is left in sb?